StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1209UnitTests.cs

```diff
@@ -0,0 +1,200 @@
+﻿namespace StyleCop.Analyzers.Test
+{
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using StyleCop.Analyzers.OrderingRules;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using TestHelper;
+    using Xunit;
+
+    public class SA1209UnitTests : CodeFixVerifier
+    {
+        [Fact]
+        public async Task TestEmptySource()
+        {
+            var testCode = string.Empty;
+            await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [Fact]
+        public async Task TestWhenAliasUsingDirectivesArePlacedCorrectly()
+        {
+            string usingsInCompilationUnit = @"using System;
+using SomeNamespace = System.IO;
+
+class A
+{
+}";
+
+            string usingsInNamespaceDeclaration = @"namespace Test
+{
+    using System;
+    using SomeNamespace = System.IO;
+
+    class A
+    {
+    }
+}";
+
+            await this.VerifyCSharpDiagnosticAsync(usingsInCompilationUnit, EmptyDiagnosticResults, CancellationToken.None);
+            await this.VerifyCSharpDiagnosticAsync(usingsInNamespaceDeclaration, EmptyDiagnosticResults, CancellationToken.None);
+        }
+
+        [Fact]
+        public async Task TestWhenUsingAliasDirectivesAreNotPlacedCorrectly()
+        {
+            var usingsInCompilationUnit = new[]
+            {
+                "namespace Xyz {}",
+                "namespace AnotherNamespace {}",
+                @"using TasksNamespace = System.Threading.Tasks;
+using Xyz;
+using System;
+using System.IO;
+using AnotherNamespace;
+class A
+{
+}"
+            };
+
+            var usingsInNamespaceDeclaration = new[]
+            {
+                "namespace Namespace {}",
+                "namespace AnotherNamespace {}",
+                @"namespace Test
+{
+    using Namespace;
+    using System.Threading;
+    using N = AnotherNamespace;
+    using System.IO;
+    using P = System.Threading.Tasks;
+    class A
+    {
+    }
+}"      };
+
+            var expectedForCompilationUnit = new[]
+            {
+                this.CSharpDiagnostic().WithLocation("Test2.cs", 1, 1).WithArguments("TasksNamespace", "AnotherNamespace"),
+            };
+
+            var expectedForNamespaceDeclaration = new[]
+            {
+                this.CSharpDiagnostic().WithLocation("Test2.cs", 5, 5).WithArguments("N", "System.Threading.Tasks"),
+            };
+
+            await this.VerifyCSharpDiagnosticAsync(usingsInCompilationUnit, expectedForCompilationUnit, CancellationToken.None);
+            await this.VerifyCSharpDiagnosticAsync(usingsInNamespaceDeclaration, expectedForNamespaceDeclaration, CancellationToken.None);
+        }
```

Comment:
I find the format of the test case pretty difficult to read, due to the multiple source files used. I don't see a direct need for the defined namespaces, as they will not influence the code under test. In my opinion they can be replaced by system namespaces without losing anything from the test cases.

I would suggest something like this:
```C#
        [Fact]
        public async Task TestWhenUsingAliasDirectivesAreNotPlacedCorrectly()
        {
            var testCodeCompilationUnit = @"using TasksNamespace = System.Threading.Tasks;
using System.Net;
using System;
using System.IO;
using System.Numerics;
class A
{
}";

            var testCodeNamespace = @"namespace Test
{
    using System.Net;
    using System.Threading;
    using N = System.Numerics;
    using System.IO;
    using P = System.Threading.Tasks;
    class A
    {
    }
}"

            DiagnosticResult[] expectedForCompilationUnit =
            {
                this.CSharpDiagnostic().WithLocation(1, 1).WithArguments("TasksNamespace", "System.Numerics")
            };

            DiagnosticResult[] expectedForNamespaceDeclaration =
            {
                this.CSharpDiagnostic().WithLocation(5, 5).WithArguments("N", "System.Threading.Tasks")
            };

            await this.VerifyCSharpDiagnosticAsync(testCodeCompilationUnit, expectedForCompilationUnit, CancellationToken.None);
            await this.VerifyCSharpDiagnosticAsync(testCodeNamespace, expectedForNamespaceDeclaration, CancellationToken.None);
        }
```